From 02230087ce3ff103a1a351593d85c8fd76aa22a8 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Wed, 4 May 2011 17:02:42 +0200
Subject: [PATCH] reindent

---
 core/ampi/SIPRegistrarClientAPI.h |   26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/core/ampi/SIPRegistrarClientAPI.h b/core/ampi/SIPRegistrarClientAPI.h
index 738279f..385282b 100644
--- a/core/ampi/SIPRegistrarClientAPI.h
+++ b/core/ampi/SIPRegistrarClientAPI.h
@@ -26,21 +26,21 @@
 
 using std::string;
 struct SIPRegistrationEvent : public AmEvent {
-	string handle;
-	unsigned int code;
-	string reason;
+  string handle;
+  unsigned int code;
+  string reason;
 	
-	SIPRegistrationEvent(int t, const string& handle,
-						 unsigned int code=0, const string& reason="") 
-		: AmEvent(t), handle(handle), code(code), reason(reason) {}
+ SIPRegistrationEvent(int t, const string& handle,
+		      unsigned int code=0, const string& reason="")
+   : AmEvent(t), handle(handle), code(code), reason(reason) {}
 
-	enum {
-		RegisterSuccess=0, 
-		RegisterFailed,
-		RegisterNoContact,
-		RegisterTimeout,
-		RegisterSendTimeout
-	};
+  enum {
+    RegisterSuccess=0,
+    RegisterFailed,
+    RegisterNoContact,
+    RegisterTimeout,
+    RegisterSendTimeout
+  };
 };
 
 const char* getSIPRegistationStateString(unsigned int s) {
-- 
1.7.10.4

