commit d1f87cfcb35f4fe27078f7760e9b4a5fff8e189c
Author: Andrew Pogrebennyk <apogrebennyk@sipwise.com>
Date:   Thu Jan 29 12:04:00 2015 +0100

    rename prompts x1..9.wav to 1..9-and.wav

--- a/apps/dsm/mods/mod_utils/ModUtils.cpp
+++ b/apps/dsm/mods/mod_utils/ModUtils.cpp
@@ -97,10 +97,10 @@
   if (right) { 
    // language has single digits before 10s
     res.push_back(basedir+int2str(num.quot * 10)+suffix);
-    res.push_back(basedir+("x"+int2str(num.rem))+suffix);    
+    res.push_back(basedir+(int2str(num.rem)+"-and")+suffix);
   } else {
     // language has single digits before 10s
-    res.push_back(basedir+("x"+int2str(num.rem))+suffix);    
+    res.push_back(basedir+(int2str(num.rem)+"-and")+suffix);
     res.push_back(basedir+int2str(num.quot * 10)+suffix);
   }
 
@@ -136,10 +136,10 @@
   if (right) { 
    // language has single digits before 10s
     sc_sess->playFile(basedir+int2str(num.quot * 10)+suffix, false);
-    sc_sess->playFile(basedir+("x"+int2str(num.rem))+suffix, false);    
+    sc_sess->playFile(basedir+(int2str(num.rem)+"-and")+suffix, false);
   } else {
     // language has single digits before 10s
-    sc_sess->playFile(basedir+("x"+int2str(num.rem))+suffix, false);    
+    sc_sess->playFile(basedir+(int2str(num.rem)+"-and")+suffix, false);
     sc_sess->playFile(basedir+int2str(num.quot * 10)+suffix, false);
   }
 
