From 95c874ef399de0a522dfdd02421e8364f2aa62bd Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 24 Nov 2011 00:11:14 +0400
Subject: [PATCH] Fix unresolved symbol in sbc module when compiled with cmake

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 apps/sbc/CMakeLists.txt |    1 +
 1 file changed, 1 insertion(+)

diff --git a/apps/sbc/CMakeLists.txt b/apps/sbc/CMakeLists.txt
index f9083b8..7201894 100644
--- a/apps/sbc/CMakeLists.txt
+++ b/apps/sbc/CMakeLists.txt
@@ -1,6 +1,7 @@
 set (sbc_SRCS
 HeaderFilter.cpp
 ParamReplacer.cpp
+RTPParameters.cpp
 RegexMapper.cpp
 SBC.cpp
 SBCCallProfile.cpp
-- 
1.7.10.4

