From d3b4db4f9597ae987384b60d5311fc86fb1f33c5 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Fri, 4 May 2012 12:56:46 +0200
Subject: [PATCH] 1.4 branch build version

---
 Makefile.defs |   28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/Makefile.defs b/Makefile.defs
index 0d2ea58..9379c0f 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -3,20 +3,20 @@ else
 makefile_defs=1
 export makefile_defs
 
-#GIT = $(shell which git)
-#SVNVERSION = $(shell which svnversion)
-#
-#ifneq ($(GIT),)
-#   SCM_REV = $(shell if git --help describe |grep dirty 2>&1 >/dev/null ; \
-#        then git describe --always --dirty 2>/dev/null; \
-#        else git describe --always 2>/dev/null; fi)
-#endif
-#
-#ifeq ($(SCM_REV),)
-#ifneq ($(SVNVERSION),)
-#   SCM_REV = r$(shell svnversion -n .)
-#endif
-#endif
+GIT = $(shell which git)
+SVNVERSION = $(shell which svnversion)
+
+ifneq ($(GIT),)
+   SCM_REV = $(shell if git --help describe |grep dirty 2>&1 >/dev/null ; \
+        then git describe --always --dirty 2>/dev/null; \
+        else git describe --always 2>/dev/null; fi)
+endif
+
+ifeq ($(SCM_REV),)
+ifneq ($(SVNVERSION),)
+   SCM_REV = r$(shell svnversion -n .)
+endif
+endif
 
 #version number
 VERSION = 1
-- 
1.7.10.4

