From b1d911f8156fb62bb0e2eb24972ed4613ee1af60 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 19 Jul 2012 14:41:01 +0400
Subject: [PATCH] [init-script] Fix diractory for Pid-file

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 pkg/rpm/sems.init |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/rpm/sems.init b/pkg/rpm/sems.init
index 15e9847..c39a721 100755
--- a/pkg/rpm/sems.init
+++ b/pkg/rpm/sems.init
@@ -24,7 +24,7 @@
 prog=sems
 sems=/usr/sbin/$prog
 
-pidfile=/var/run/$prog.pid
+pidfile=/var/run/$prog/$prog.pid
 lockfile=/var/lock/subsys/$prog
 conffile=/etc/$prog/$prog.conf
 
-- 
1.7.10.4

