From c314864703da53157528a53ac035e0f2968dd525 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 19 Jul 2012 14:42:01 +0400
Subject: [PATCH] [systemd] Restart in case of failure

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 pkg/rpm/sems.systemd.service |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/pkg/rpm/sems.systemd.service b/pkg/rpm/sems.systemd.service
index 7d66886..43996a5 100644
--- a/pkg/rpm/sems.systemd.service
+++ b/pkg/rpm/sems.systemd.service
@@ -8,6 +8,9 @@ User=sems
 Group=sems
 EnvironmentFile=-/etc/sysconfig/sems
 PIDFile=/var/run/sems/sems.pid
+Restart=always
+StartLimitInterval=10
+StartLimitBurst=5
 ExecStart=/usr/sbin/sems -E -u sems -g sems -P /var/run/sems/sems.pid $OPTIONS
 
 [Install]
-- 
1.7.10.4

