From caf612aae5c09ea64fc62032920abb61af347311 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Sun, 15 Apr 2012 21:06:55 +0200
Subject: [PATCH] DSM: safeguard transferOwnership/releaseOwnership agains
 NULL ptrs

---
 apps/dsm/DSMCall.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/apps/dsm/DSMCall.cpp b/apps/dsm/DSMCall.cpp
index d48f2fc..5ae7fd3 100644
--- a/apps/dsm/DSMCall.cpp
+++ b/apps/dsm/DSMCall.cpp
@@ -619,10 +619,14 @@ void DSMCall::addSeparator(const string& name, bool front) {
 }
 
 void DSMCall::transferOwnership(DSMDisposable* d) {
+  if (d == NULL)
+    return;
   gc_trash.insert(d);
 }
 
 void DSMCall::releaseOwnership(DSMDisposable* d) {
+  if (d == NULL)
+    return;
   gc_trash.erase(d);
 }
 
-- 
1.7.10.4

