From 3d02aac5a02caf6ba9cba8e03a8f4adb5b16f9f6 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Mon, 16 Apr 2012 16:36:41 +0200
Subject: [PATCH] excluding mobile_push app from making

Conflicts:

	apps/Makefile
---
 apps/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/Makefile b/apps/Makefile
index f947059..394a78f 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -23,7 +23,7 @@ else
  ifneq ($(local_exclude_app_modules),)
    exclude_app_modules = $(local_exclude_app_modules)
  else
-   exclude_app_modules += mp3 examples py_sems xmlrpc2di gateway jsonrpc twit
+   exclude_app_modules += mp3 examples py_sems xmlrpc2di gateway jsonrpc twit mobile_push
 
    ifneq ($(USE_MONITORING), yes)
    exclude_app_modules += monitoring
-- 
1.7.10.4

