From 1057989d113c29bc2f509f6e2d2d887f1f610341 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Tue, 17 Apr 2012 19:22:06 +0200
Subject: [PATCH] mobile_push: handle CANCEL properly

---
 apps/mobile_push/mobile_push.dsm |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/apps/mobile_push/mobile_push.dsm b/apps/mobile_push/mobile_push.dsm
index 938971f..61ad783 100644
--- a/apps/mobile_push/mobile_push.dsm
+++ b/apps/mobile_push/mobile_push.dsm
@@ -185,4 +185,9 @@ transition "timer hit" WAITING - timer(#id==1) / {
   stop(false);
 } -> END;
 
+transition "CANCEL received" WAITING - hangup / {
+  subscription.remove($r.handle);
+  stop(false);
+} -> END;
+
 state END;
\ No newline at end of file
-- 
1.7.10.4

