From 2bbd69c5a92bf6c2403dd69f572b78567e8056f9 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Thu, 19 Apr 2012 12:11:33 +0200
Subject: [PATCH] mobile-push: b/f: reply 487 to INVITE on CANCEL

---
 apps/mobile_push/mobile_push.dsm |    1 +
 1 file changed, 1 insertion(+)

diff --git a/apps/mobile_push/mobile_push.dsm b/apps/mobile_push/mobile_push.dsm
index 0333513..824126b 100644
--- a/apps/mobile_push/mobile_push.dsm
+++ b/apps/mobile_push/mobile_push.dsm
@@ -196,6 +196,7 @@ transition "timer hit" WAITING - timer(#id==1) / {
 
 transition "CANCEL received" WAITING - hangup / {
   subscription.remove($r.handle);
+  dlg.reply(487, "Request Terminated");
   stop(false);
 } -> END;
 
-- 
1.7.10.4

