From 99dbc30fa75456c18700830ac4b08d4b6049d984 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Thu, 1 Aug 2013 19:03:07 +0200
Subject: [PATCH] core: AmSipMsg gets ArgObject as base (for passing in DI
 calls)

---
 core/AmSipMsg.h |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/AmSipMsg.h b/core/AmSipMsg.h
index a57e0f6..f54e0c7 100644
--- a/core/AmSipMsg.h
+++ b/core/AmSipMsg.h
@@ -5,9 +5,11 @@
 using std::string;
 
 #include "sip/trans_layer.h"
+#include "AmArg.h"
 
 /* enforce common naming in Req&Rpl */
 class _AmSipMsgInDlg
+  : public ArgObject
 {
  public:
   string       method;
-- 
1.7.10.4

