From 26035b6f2665249bd6831b680488595755478108 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Mon, 19 Aug 2013 18:03:22 +0200
Subject: [PATCH] sbc: update SBCCallRegistry only on positive replies to
 INVITE

---
 apps/sbc/SBC.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/sbc/SBC.cpp b/apps/sbc/SBC.cpp
index b589fb5..ad8a055 100644
--- a/apps/sbc/SBC.cpp
+++ b/apps/sbc/SBC.cpp
@@ -1339,7 +1339,7 @@ void SBCCalleeSession::onSipReply(const AmSipReply& reply, int old_dlg_status,
 
   // update call registry (unfortunately has to be done always -
   // not possible to determine if learned in this reply)
-  if (!dlg.remote_tag.empty()) {
+  if (!dlg.remote_tag.empty() && reply.code >= 200 && trans_method == SIP_METH_INVITE) {
     SBCCallRegistry::updateCall(other_id, dlg.remote_tag);
   }
 
-- 
1.7.10.4

