From 6e77a3afe61d8a3036f0941b65f8e067034c5ff8 Mon Sep 17 00:00:00 2001
From: Stefan Sayer <stefan.sayer@googlemail.com>
Date: Mon, 19 Aug 2013 18:21:55 +0200
Subject: [PATCH] sbc: call registry: more verbosity on new entry

---
 apps/sbc/SBCCallRegistry.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/sbc/SBCCallRegistry.cpp b/apps/sbc/SBCCallRegistry.cpp
index c0677c4..8d116fd 100644
--- a/apps/sbc/SBCCallRegistry.cpp
+++ b/apps/sbc/SBCCallRegistry.cpp
@@ -34,7 +34,7 @@ void SBCCallRegistry::addCall(const string& ltag, const SBCCallRegistryEntry& ot
   registry[ltag] = other_dlg;
   registry_mutex.unlock();
 
-  DBG("SBCCallRegistry: Added call '%s' - mapped to: '%s'\n", ltag.c_str(), other_dlg.ltag.c_str());
+  DBG("SBCCallRegistry: Added call '%s' - mapped to: '%s'/'%s'/'%s'\n", ltag.c_str(), other_dlg.ltag.c_str(), other_dlg.rtag.c_str(), other_dlg.callid.c_str());
 }
 
 void SBCCallRegistry::updateCall(const string& ltag, const string& other_rtag) {
-- 
1.7.10.4

