diff --git a/core/AmSipDispatcher.cpp b/core/AmSipDispatcher.cpp
index 574ce54..d6dc107 100644
--- a/core/AmSipDispatcher.cpp
+++ b/core/AmSipDispatcher.cpp
@@ -86,7 +86,7 @@ void AmSipDispatcher::handleSipMsg(AmSipRequest &req)
   }
 
   DBG("method: `%s' [%zd].\n", req.method.c_str(), req.method.length());
-  if(req.method == "INVITE"){
+  if(req.method == "INVITE" || req.method == "NOTIFY"){
       
       AmSessionContainer::instance()->startSessionUAS(req);
   }
