From 1619876f76da9ac3530b3524795ba06ddcfdab09 Mon Sep 17 00:00:00 2001
From: Roman Romanchenko <rromanchenko@sipwise.com>
Date: Thu, 29 Dec 2022 10:14:26 +0100
Subject: MT#55831 sems: fails to build against Debian/bullseye

- python support removed
(real ticket number: TT#101059)

--- a/Makefile.defs
+++ b/Makefile.defs
@@ -85,7 +85,7 @@
 # build in support for monitoring?
 #
 #
-USE_MONITORING=yes
+USE_MONITORING=no
 
 # Support for long debug messages? (useful for debugging SIP messages' contents)
 #
