From e580ff5ca6db1d455d6dc8e8942b8f29e0913ca7 Mon Sep 17 00:00:00 2001
From: Victor Seva <vseva@sipwise.com>
Date: Thu, 29 Dec 2022 10:14:06 +0100
Subject: MT#55831 No config patch

No config patch

--- a/apps/Makefile
+++ b/apps/Makefile
@@ -69,7 +69,7 @@
 	done 
 
 .PHONY: install
-install: install-bin install-cfg
+install: install-bin
 
 .PHONY: install-bin
 install-bin:
--- a/apps/dsm/mods/Makefile
+++ b/apps/dsm/mods/Makefile
@@ -45,7 +45,7 @@
 	done 
 
 .PHONY: install
-install: install-bin install-cfg
+install: install-bin
 
 .PHONY: install-bin
 install-bin:
--- a/apps/examples/Makefile
+++ b/apps/examples/Makefile
@@ -33,7 +33,7 @@
 	done 
 
 .PHONY: install
-install: install-bin install-cfg
+install: install-bin
 
 .PHONY: install-bin
 install-bin:
--- a/apps/examples/tutorial/Makefile
+++ b/apps/examples/tutorial/Makefile
@@ -33,7 +33,7 @@
 	done 
 
 .PHONY: install
-install: install-bin install-cfg
+install: install-bin
 
 .PHONY: install-bin
 install-bin:
--- a/core/Makefile
+++ b/core/Makefile
@@ -117,8 +117,6 @@
 
 install: all mk-install-dirs \
 	install-audio \
-	install-modules-cfg \
-	install-cfg \
 	install-bin \
 	install-modules
 
